data/logs_model/logs_producer/splunk_logs_producer.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from splunklib import client
+from splunklib.binding import HTTPError
+
+from data.logs_model.logs_producer.interface import LogProducerInterface
+
+logger = logging.getLogger(__name__)
+
+
+class SplunkLogsProducer(LogProducerInterface):
+    """
+    Log producer for writing log entries to Splunk
+    This implementation writes directly to Splunk without a streaming/queueing service.
+    """
+
+    def __init__(
+        self,
+        host,
+        port,
+        bearer_token=None,
+        use_ssl=True,
+        index_prefix=None,
+    ):
+        try:
+            service = client.connect(host=host, port=port, token=bearer_token, verify=use_ssl)
+        except Exception as e:
+            logger.exception("Failed to connect to Splunk instance %s", e)
+        else:
+            try:
+                self.index = service.indexes[index_prefix]
+                logger.info("splunk index %s", self.index)
+            except Exception as e:
+                self.index = service.indexes.create(index_prefix)
+
+    def send(self, log):
+        try:
+            self.index.submit(log, sourcetype="access_combined", host="quay")
+        except Exception as e:
+            logger.exception("SplunkLogsProducer exception sending log to Splunk: %s", e)
```

Comment:
Same question here. When the `submit` call above raises an exception you are logging it and nothing else.
If we are using the elastic logs producer as reference, when it encounters an exception it both logs it and raises another exception of it's own (`LogSendException`).